packages/api/src/middleware/validation/organicHistoryLocationCheck.js

```diff
@@ -0,0 +1,16 @@
+const locationModel = require('../../models/locationModel');
+const organicHistoryLocations = [
+  'field', 'garden', 'greenhouse',
+];
+
+async function organicHistoryLocationCheck(req, res, next) {
+  const location = await locationModel.query().join('figure', 'figure.location_id', 'location.location_id')
+    .where('location.location_id', req.body.location_id)
+    .whereIn('type', organicHistoryLocations);
+  if (!location.length) return res.status(400).send('Location must be crop enabled.');
```

Comment:
validation should be in middleware

Reply:
I'm ok with this for now, but apart from LF convention I don't think that validation must necessarily go in middleware. This change complicates the codebase without gaining anything. I don't think there's anyplace else this would be re-used, and if there were it would likely be more useful as "crop enabled".